src/main/java/org/threadly/concurrent/future/SettableListenableFuture.java

```diff
@@ -141,6 +143,57 @@ public void setFailure(Throwable failure) {
   }
   
   /**
+   * Optional call to set the thread internally that will be generating the result for this 
+   * future.  Setting this thread allows it so that if a {@link #cancel(boolean)} call is invoked 
+   * with {@code true}, we can send an interrupt to this thread.  
+   * 
+   * The reference to this thread will be cleared after this future has completed (thus allowing 
+   * it to be garbage collected).
+   * 
+   * @param thread Thread that is generating the result for this future
+   */
+  public void setRunningThread(Thread thread) {
+    if (! done) {
+      this.runningThread = thread;
+    }
+  }
+
+  @Override
+  public boolean cancel(boolean interruptThread) {
+    boolean canceled;
```

Comment:
I would rename this local variable, makes following which one is set for what confusing, maybe didCancel or something like that